packages/material-ui-styles/src/withTheme.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import getDisplayName from '@material-ui/utils/getDisplayName';
+import { ThemeContext } from './ThemeProvider';
+
+// Provide the theme object as a property to the input component.
+const withTheme = () => Component => {
+  const WithTheme = props => (
+    <ThemeContext.Consumer>
+      {theme => {
+        const { innerRef, ...other } = props;
+        return <Component theme={theme} ref={innerRef} {...other} />;
+      }}
+    </ThemeContext.Consumer>
+  );
+
+  WithTheme.propTypes = {
+    /**
+     * Use that property to pass a ref callback to the decorated component.
+     */
+    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
```

Comment:
Same as with `withStyles` but `withTheme` doesn't use `forwardRef` with this implementation.